fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api;
+
+/**
+ * A collection of RDF contexts that can be used to extract triples from FedoraResources
+ *
+ * @author acoburn
+ * @since Dec 4, 2015
+ */
+public enum RdfContext {
```

Comment:
Yes, this is all why I would rather not use the `enum` here, although I appreciate that @acoburn is just trying to get information passed around in a reasonable way. The context is: @acoburn is redoing RDF generation mechanics to use the new Java 8 Stream API, which is the general-purpose and well-supported version of what @cbeer and I did when we first built this stuff. In order to do that, he wants (quite reasonably) to provide some means by which clients can indicate what kind of RDF they want. There is no point to generating RDF and then filtering it, and it could be very expensive for some implementations. That's where this comes in. We _were_ passing around actual values for classes (`Foo.class`) but that couples the client directly to impls (boo!). The ideal,  to my mind, would be to develop a good set of interface types that actually model the RDF generation flavors we expect to support, and then pass those around as types. No coupling, extensible, no problem. But that woul admittedly slow this down and be more work for @acoburn.